idaes/power_generation/carbon_capture/mea_solvent_system/unit_models/phe.py

```diff
@@ -0,0 +1,801 @@
+# software owners: The Regents of the University of California, through
+# Lawrence Berkeley National Laboratory,  National Technology & Engineering
+# Solutions of Sandia, LLC, Carnegie Mellon University, West Virginia
+# University Research Corporation, et al. All rights reserved.
+#
+# Please see the files COPYRIGHT.txt and LICENSE.txt for full copyright and
+# license information, respectively. Both files are also available online
+# at the URL "https://github.com/IDAES/idaes-pse".
+##############################################################################
+"""
+IDAES PLATE HEAT EXCHANGER MODEL (PHE) USING EFFECTIVENESS-NTU METHOD
+
+Schematic diagram
+
+hot fluid----->|                                                   |<--- cold fluid
+(even channels)|      Pass 1   |     Pass 2 to P-1 |    Pass P     |(odd channels)
+               |    ___     ___|                   |    ___     ___|
+               |   |   |   |   |                   |   |   |   |   |
+               |   |   |   |   |                   |   |   |   |   |
+               |   |   |   |   |                   |   |   |   |   |
+               |   |___|   |___|                   |   |___|   |___|
+               |               |                   |               |
+channels         1   2   3  Np                       1   2   3  Np
+
+P  : number of passes(same for both fluids)
+if P is even: PHE is in counter-current mode
+if P is odd : PHE  is in co-current mode
+Np : number of channels in a pass(should be even for balanced PHE)
+Hot and cold fluids flow alternatively in the channels.
+
+Note: Total Heat transfer area of the PHE can be increased by adding more plates,
+which is equilvalent to increasing the number of channels in a pass(Np)
+
+Detailed model equations can be found in the paper :
+Akula, P., Eslick, J., Bhattacharyya, D. and Miller, D.C., 2019.
+Modelling and Parameter Estimation of a Plate Heat Exchanger
+as Part of a Solvent-Based Post-Combustion CO2 Capture System.
+In Computer Aided Chemical Engineering (Vol. 47, pp. 47-52). Elsevier.
+"""
+
+# Import Pyomo libraries
+from pyomo.environ import Param, RangeSet, Constraint, Expression,\
+    SolverFactory, value, Var, exp, units as pyunits
+from pyomo.common.config import ConfigBlock, ConfigValue, In
+
+# Import IDAES cores
+from idaes.core import (ControlVolume0DBlock,
+                        declare_process_block_class,
+                        MaterialBalanceType,
+                        EnergyBalanceType,
+                        MomentumBalanceType,
+                        UnitModelBlockData,
+                        useDefault)
+from idaes.core.util.config import is_physical_parameter_block
+from idaes.core.util.exceptions import ConfigurationError
+from idaes.core.util.misc import add_object_reference
+import idaes.logger as idaeslog
+
+__author__ = "Paul Akula"
+
+
+# Set up logger
+_log = idaeslog.getLogger(__name__)
+
+
+class plate_param(object):
```

Comment:
Is this really necessary as a separate object? Use of this is hard-coded into parameter creation, so I don't see a lot of advantage to this at the moment. However, if an object like this (or better yet a dict) were to be used as a configuration argument, then this could be useful.